addons/l10n_in_website_sale/static/src/js/portal.js

```diff
@@ -0,0 +1,86 @@
+import publicWidget from "@web/legacy/js/public/public_widget";
+import { debounce } from "@web/core/utils/timing";
+
+publicWidget.registry.portalDetailsExtend = publicWidget.registry.portalDetails.extend({
+    events: {
+        'input input[name="vat"]': "_onChangeVat",
+        'change select[name="country_id"]': "_onCountryChange",
+    },
+
+    init: function () {
+        this._super.apply(this, arguments);
+
+        this._onChangeVat = debounce(this._onChangeVat.bind(this), 400);
+        this.countryCode = document.querySelector(
+            'form[action="/my/account"]'
+        ).dataset.companyCountryCode;
+        this.l10n_in_gst_treatment = document.getElementById("l10n_in_gst_treatment_container");
+        this._gst_treatment = document.getElementById("l10n_in_gst_treatment")?.value ?? "";
+        this._invalid_vat = false;
+
+        this._setL10nInGstTreatment();
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onChangeVat(ev) {
+        if (this.countryCode === "IN") {
+            this._setL10nInGstTreatment();
+        }
+    },
+
+    /**
+     * @override
+     */
+    _onCountryChange: function () {
+        if (this.countryCode === "IN") {
+            this._setL10nInGstTreatment();
+        }
+        return this._super();
+    },
+
+    _setL10nInGstTreatment() {
+        let selectedCountry = document.getElementsByName("country_id")[0];
+        selectedCountry = selectedCountry.options[selectedCountry.selectedIndex].innerText.trim();
```

Comment:
See the changes in template, we can now access it using this.el
`const selectedCountry = this.el.querySelector('select[name="country_id"]').dataset.companyCountryCode; `